src/components/welcome-screen/welcome-screen.tsx

```diff
@@ -1,21 +1,46 @@
 import {h} from 'preact';
 import {KalturaNullableBoolean} from '../../providers/response-types';
 import * as styles from './welcome-screen.scss';
+import {QuizTranslates} from '../../types';
+import {icons} from '../icons';
+const {withText, Text} = KalturaPlayer.ui.preacti18n;
+const {Icon} = KalturaPlayer.ui.components;
 
-const {Overlay} = KalturaPlayer.ui.components;
+const translates: QuizTranslates = {
+  welcomeTitle: <Text id="ivq.welcome_title">Start your video quiz!</Text>,
+  welcomeDownload: <Text id="ivq.welcome_download">Download Pre-Test</Text>
+};
 
 interface WelcomeScreenProps {
   allowDownload?: KalturaNullableBoolean;
   welcomeMessage?: string;
+  onDownload: () => void;
   onClose: () => void;
 }
 
-export const WelcomeScreen = ({allowDownload, welcomeMessage, onClose}: WelcomeScreenProps) => {
+export const WelcomeScreen = withText(translates)(({allowDownload, welcomeMessage, onDownload}: WelcomeScreenProps) => {
```

Comment:
here you use translates from your const. Instead you should use translates from component props:
`export const WelcomeScreen = withText(translates)(({allowDownload, welcomeMessage, onDownload, ...translates}: WelcomeScreenProps) => {`

Reply:
done